include/taskolib/exceptions.h

```diff
@@ -98,6 +98,14 @@ class Error : public std::runtime_error
         return !(lhs == rhs);
     }
 
+    /**
+     * Set the associated step index.
+     *
+     * The step index may be set to nullopt to indicate that the error is not associated
+     * with a specific step.
+     */
+    void set_index(OptionalStepIndex opt_step_index) { index_ = opt_step_index; }
```

Comment:
Because we use an `Error` object to store the error information from the last sequence execution. We could of course replace the entire object instead of manipulating just the index inside... would you prefer that?

Reply:
I would prefer that. What do @Finii and @wyrnat think about that? Any opinion?

Reply:
I have removed the commit introducing `set_index()`. Instead, we now assign an entirely new `Error` object with a copy of the original message:
```cpp
    error_ = Error(error_.value().what(), get_new_index(*maybe_error_idx));
```
Is this better? 🤷